### ISSUE TITLE
Missing Template

### ISSUE THREAD

**Issue by steveroot** [spree-contrib/spree_paypal_express#49]
Having followed the install instructions on github, I get a 'Missing Template' error.
I can see the files in the github repo and rails appears to be searching all the correct places. I can also see the file locally.

---below this, error page, locating files in terminal, then Gemfile.lock spree sections--
#### Error

```
Missing partial spree/checkout/payment/paypalexpressuk with {:locale=>[:"en-GB", :en], :formats=>[:html], :handlers=>[:erb, :builder, :coffee, :rabl]}. Searched in:
  * "/Users/steve/Documents/ruby/rkbbspree/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree_paypal_express-11c8935aac1d/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree_static_content-2b1a3fa42581/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree-498ab18ff74b/promo/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree-498ab18ff74b/dash/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree-498ab18ff74b/api/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree-498ab18ff74b/auth/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/gems/devise-2.0.4/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree-498ab18ff74b/core/app/views"
  * "/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/gems/kaminari-0.13.0/app/views"
```
#### Finding the files in terminal:

1.9.3@rkbbspree steve@Steve-Root-MBP:payment$pwd
/Users/steve/.rvm/gems/ruby-1.9.3-p125@rkbbspree/bundler/gems/spree_paypal_express-11c8935aac1d/app/views/spree/checkout/payment
1.9.3@rkbbspree steve@Steve-Root-MBP:payment$ls
_paypalexpress.html.erb
#### Gemfile.lock

GIT
  remote: git://github.com/spree/spree.git
  revision: 34f2e319ad81f301285366bc523ec75332fafea5
  branch: 1-1-stable
  specs:
    spree (1.1.0.rc2)
      spree_api (= 1.1.0.rc2)
      spree_auth (= 1.1.0.rc2)
      spree_cmd (= 1.1.0.rc2)
      spree_core (= 1.1.0.rc2)
      spree_dash (= 1.1.0.rc2)
      spree_promo (= 1.1.0.rc2)
      spree_sample (= 1.1.0.rc2)
    spree_api (1.1.0.rc2)
      rabl (= 0.6.5)
      spree_auth (= 1.1.0.rc2)
      spree_core (= 1.1.0.rc2)
    spree_auth (1.1.0.rc2)
      cancan (= 1.6.7)
      devise (~> 2.0)
      spree_core (= 1.1.0.rc2)
    spree_cmd (1.1.0.rc2)
    spree_core (1.1.0.rc2)
      activemerchant (= 1.20.4)
      acts_as_list (= 0.1.4)
      aws-sdk (~> 1.3.4)
      deface (>= 0.8.0)
      ffaker (~> 1.12.0)
      highline (= 1.6.11)
      jquery-rails (~> 2.0.0)
      kaminari (>= 0.13.0)
      nested_set (= 1.7.0)
      paperclip (~> 2.7)
      rails (>= 3.2.2, <= 3.2.3)
      ransack (~> 0.6.0)
      state_machine (= 1.1.2)
      stringex (~> 1.3.2)
    spree_dash (1.1.0.rc2)
      httparty (~> 0.8.1)
      spree_auth (= 1.1.0.rc2)
      spree_core (= 1.1.0.rc2)
    spree_promo (1.1.0.rc2)
      spree_auth (= 1.1.0.rc2)
      spree_core (= 1.1.0.rc2)
    spree_sample (1.1.0.rc2)
      spree_core (= 1.1.0.rc2)

GIT
  remote: git://github.com/spree/spree_i18n.git
  revision: 00cd5ed517fd608d8df41964383726035bed2203
  specs:
    spree_i18n (1.0.0)
      spree_core (>= 0.30.0)

GIT
  remote: git://github.com/spree/spree_paypal_express.git
  revision: 11c8935aac1d78aa99e0b23e005f5c98ecea05f2
  specs:
    spree_paypal_express (1.1.0)
      spree_core (>= 1.0.0)

GIT
  remote: git://github.com/steveroot/spree_static_content
  revision: 2b1a3fa425814bf13546269e7409ff01530cd066
  branch: master
  specs:
    spree_static_content (1.1.0.beta)
      spree_auth (~> 1.1.0.beta)
      spree_core (~> 1.1.0.beta)

**Comment by steveroot** [comment]
More detail from the error:

```
14: <% @order.available_payment_methods.each do |method| %>
15:         <li id="payment_method_<%= method.id %>" class="<%= 'last' if method == @order.available_payment_methods.last %>" data-hook>
16:           <fieldset>
17: <%= render :partial => "spree/checkout/payment/#{method.method_type}", :locals => { :payment_method => method } %>
18:           </fieldset>
19: </li>
20:       <% end %>
```

and a possibly relevant post on StackOverflow: http://stackoverflow.com/questions/9936675/missing-partial-when-using-devise-2-0-w-rails-3-1-3

Extract from question: ...My app was working fine and rendering the cart partial in every view. But when I installed devise 2.0, the partial could no longer be found for devise views....

Accepted Answer: You should be able to specify the partial with just:
= render :partial => 'carts/cart', :object => @cart
The views/ part of your definition is probably throwing it off. app/views is implied, so when you specify views/carts/cart, it's probably not finding a views directory under app/views.

**Comment by radar** [comment]
It's looking for paypalexpressuk but only finding paypalexpress. Missing the "uk" at the end.

**Comment by steveroot** [comment]
Thanks Radar, just noticed that myself. Closed it with a comment that seems to have vanished - I couldn't find the 'uk' bit within the code so I think that I must have (once again) broken something without realising. 
Thanks for the help!

**Comment by steveroot** [comment]
OK, so I've had time to return to this.
In the spree admin console, there are two providers generated by the extension:
Spree::BillingIntegration::PaypalExpress
Spree::BillingIntegration::PaypalExpressUK
I selected the UK option, which generates the missing template error.  Selecting the first option doesn't give the error.  I'm not sure what the UK option does yet...

**Comment by radar** [comment]
It's likely that the two are very similar. Could you try forking this extension, copying over the typical paypal_express extension into the location of the missing template and seeing if that works?

**Comment by steveroot** [comment]
Sure. I'm going to try and get the non-uk version working first, then I'll come back and investigate further (probably won't be in the next few days)

**Comment by steveroot** [comment]
I haven't got the non-uk version working yet (for an unrelated problem), so as a break I thought I'd manage to get the UK version at least to the same point of failure.  

So, I copied the missing template and that brought a new error,
which appeared to related to variables in the model, so i copied those across but I'm now left with
undefined method `preferred_currency' for #Spree::BillingIntegration::PaypalExpressUk:0x007fc2cfcc21f0
Which appears to me to be related to the checkout_controller_decorator.rb but fixing this appears to be beyond my skillset. Forked at: git@github.com:steveroot/spree_paypal_express.git in case it's obvious to anyone else.

Thanks for any pointers!

**Comment by radar** [comment]
I've fixed this now on the repository with @f7b2775. There were a couple of other things I ran into in my testing too that I've also fixed. Let me know if this now works for you too :)
